HyperVExtension/src/HyperVExtension/Helpers/DevSetupAgentDeploymentHelper.cs

```diff
@@ -0,0 +1,306 @@
+        var result = _powerShellService.Execute(deployDevSetupAgentStatement, PipeType.DontClearBetweenStatements);
+        if (!string.IsNullOrEmpty(result.CommandOutputErrorMessage))
+        {
+            throw new DevSetupAgentDeploymentException(
+                $"Unable to deploy DevSetupAgent service to VM with Id: {_vmId} due to PowerShell error: {result.CommandOutputErrorMessage}");
+        }
+    }
+
+    public virtual string GetSourcePath(ushort architecture)
+    {
+        if (architecture == (ushort)ProcessorArchitecture.X64)
+        {
+            // TODO: Adjust path once we have the actual DevSetupAgent package.
+            return Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "Assets", "DevSetupAgent_x64.zip");
+        }
+        else
+        {
+            throw new DevSetupAgentDeploymentException(
+                $"Unable to deploy DevSetupAgent service to VM with Id: {_vmId} due to unsupported architecture: {architecture}");
+        }
+    }
+
+    public PSObject GetSessionObject(PSCredential credential)
+    {
+        var newSessionCommand = new StatementBuilder()
+            .AddCommand("New-PSSession")
+            .AddParameter("VMId", _vmId)
+            .AddParameter("Credential", credential)
+            .Build();
+
+        var result = _powerShellService.Execute(newSessionCommand, PipeType.None);
+        if (!string.IsNullOrEmpty(result.CommandOutputErrorMessage))
+        {
+            throw new DevSetupAgentDeploymentSessionException(
+                $"Unable to create remote session for VM with Id: {_vmId} due to PowerShell error: {result.CommandOutputErrorMessage}");
+        }
+
+        return result!.PsObjects.FirstOrDefault()!;
+    }
+
+    private ushort GetVMArchitechture(PSObject session)
+    {
+        var getVMArchitechtureCommand = new StatementBuilder()
+                       .AddCommand("Invoke-Command")
+                       .AddParameter("Session", session)
+                       .AddParameter("ScriptBlock", ScriptBlock.Create("(Get-CIMInstance -Class win32_processor).Architecture"))
+                       .Build();
+
+        var result = _powerShellService!.Execute(getVMArchitechtureCommand, PipeType.None);
+        if (!string.IsNullOrEmpty(result.CommandOutputErrorMessage))
+        {
+            throw new DevSetupAgentDeploymentException(
+                               $"Unable to get VM architecture for VM with Id: {_vmId} due to PowerShell error: {result.CommandOutputErrorMessage}");
+        }
+
+        var psObject = result.PsObjects.FirstOrDefault();
+        if (psObject == null)
+        {
+            throw new DevSetupAgentDeploymentException(
+                               $"Unable to get VM architecture for VM with Id: {_vmId} due to PowerShell error: No result returned");
+        }
+
+        return (ushort)psObject.BaseObject;
+    }
+
+    private readonly string _script = @"
+function DeployDevSetupAgent

```

Comment:
Powershell has a naming convention for function names where the function name should start with an approved verb. So in this case I believe we should rename it to:

```Suggestion
function Publish-DevSetupAgent
```
check the "Deploy" case here: https://learn.microsoft.com/en-us/powershell/scripting/developer/cmdlet/approved-verbs-for-windows-powershell-commands?view=powershell-7.4#data-verbs

 https://learn.microsoft.com/en-us/powershell/scripting/learn/ps101/09-functions?view=powershell-7.4#naming

Reply:
Originally, I had Add-DevSetupAgent cmdlet which didn't work with C#. And this code is basically an "internal" function for that cmdlet.
I think "Install" is a good choice here. It's paired with "Uninstall".